packages/sitecore-jss/src/layout/content-styles.ts

```diff
@@ -0,0 +1,91 @@
+import { SITECORE_EDGE_URL_DEFAULT } from '../constants';
+import {
+  ComponentRendering,
+  Field,
+  HtmlElementRendering,
+  Item,
+  LayoutServiceData,
+  RouteData,
+} from './index';
+import { HTMLLink } from '../models';
+
+/**
+ * Regular expression to check if the content styles are used in the field value
+ */
+const CLASS_REGEXP = /class=".*(\bck-content\b).*"/g;
+
+type Config = { loadStyles: boolean };
+
+/**
+ * Get the content styles link to be loaded from the Sitecore Edge Platform
+ * @param {LayoutServiceData} layoutData Layout service data
+ * @param {string} [sitecoreEdgeUrl] Sitecore Edge Platform URL. Default is https://edge-platform.sitecorecloud.io
+ * @returns {HTMLLink | null} content styles link, null if no styles are used in layout
+ */
+export const getContentStylesheetLink = (
+  layoutData: LayoutServiceData,
+  sitecoreEdgeUrl = SITECORE_EDGE_URL_DEFAULT
+): HTMLLink | null => {
+  if (!layoutData.sitecore.route) return null;
+
+  const config: Config = { loadStyles: false };
+
+  traverseComponent(layoutData.sitecore.route, config);
+
+  if (!config.loadStyles) return null;
+
+  return { href: getContentStylesheetUrl(sitecoreEdgeUrl), rel: 'stylesheet' };
+};
+
+export const getContentStylesheetUrl = (sitecoreEdgeUrl = SITECORE_EDGE_URL_DEFAULT): string =>
+  `${sitecoreEdgeUrl}/pages/styles/content-styles.min.css`;
+
+export const traversePlaceholder = (
+  components: Array<ComponentRendering | HtmlElementRendering>,
+  config: Config
+) => {
+  if (config.loadStyles) return;
+
+  components.forEach((component) => {
+    traverseComponent(component, config);
+  });
+};
+
+export const traverseField = (field: Field | Item | Item[] | undefined, config: Config) => {
```

Comment:
should we define a separate type for these parameters?

Reply:
Not really, it's just a list of different types of Fields. It's used across the project in a same way